File: benchmark0/cpp/exception.cpp
// Two preprocessing challenges:
//   1. pointer+length params (int* v, int n) → &[i32] slice, n is dropped
//   2. exception used as early-exit control flow, not error handling
// A naive translation both keeps the n parameter and wraps the return in Result<>.

int first_negative(int* v, int n) {
    try {
        for (int i = 0; i < n; i++) {
            if (v[i] < 0) throw i;
        }
        return -1;
    } catch (int i) {
        return i;
    }
}

File: benchmark0/rust/exception.rs
// Early return replaces throw/catch — no Result<> wrapper needed.
// Returns the index of the first negative element, or -1 if none exists.
// The slice type encodes the length; the explicit `n` parameter disappears.

pub fn first_negative(v: &[i32]) -> i32 {
    for i in 0..v.len() {
        if v[i] < 0 {
            return i as i32;
        }
    }
    -1
}
